tests/test_product.py

```diff
@@ -0,0 +1,51 @@
+import pytest
+from buy_together_app import models
+
+QR = "Q5o76MbdiNbXprNEnHfpcGWFp1CMF8XY"
+NAME = "Banana"
+DESCRIPTION = "Good!"
+
+
+@pytest.fixture
+def product0():
+    return models.Product(qr_code=QR, product_name=NAME, description=DESCRIPTION)
+
+
+@pytest.fixture
+def saved_product0(product0):
+    product0.save()
+    return product0
+
+
+class TestProductModel:
+    def test_new_product(self, product0):
+        assert product0.qr_code == QR
+        assert product0.product_name == NAME
+        assert product0.description == DESCRIPTION
+
+    @pytest.mark.django_db()
+    def test_save_product(self, saved_product0):
+        assert saved_product0 in models.Product.objects.all()
+
+    @pytest.mark.django_db()
+    def test_delete_product(self, saved_product0):
+        assert saved_product0 in models.Product.objects.all()
+        saved_product0.delete()
+        assert saved_product0 not in models.Product.objects.all()
+
+    @pytest.mark.django_db()
+    def test_filter_by_qr(self, saved_product0):
+        assert [saved_product0] == list(models.Product.filter_qr(saved_product0.qr_code))
+
+    @pytest.mark.django_db()
+    def test_filter_by_product_name(self, saved_product0):
+        assert saved_product0 in list(models.Product.filter_name(saved_product0.product_name))
+
+    @pytest.mark.django_db()
+    def test_filter_by_description(self, saved_product0):
+        assert saved_product0 in models.Product.filter_description(saved_product0.description)
```

Comment:
this will not fail even if the function returns all products for some reason. I think the approach in `test_filter_by_qr` is more accurate.

Reply:
We can have multiple descriptions that is the same,qr is unique

Reply:
same answer like before